.github/workflows/test.yaml

```diff
@@ -1,52 +1,47 @@
-use clap::{Parser, Subcommand};
-use std::fs::{read_to_string, OpenOptions};
-use std::io::Write;
-
-#[derive(Parser)]
-#[command(name = "Todo")]
-#[command(about = "A simple to-do list CLI app")]
-struct Cli {
-    #[command(subcommand)]
-    command: Commands,
-}
-
-#[derive(Subcommand)]
-enum Commands {
-    Add {
-        #[arg()]
-        task: String,
-    },
-    List,
-}
-
-fn main() {
-    let cli = Cli::parse();
-
-    match &cli.command {
-        Commands::Add { task } => add_task(task),
-        Commands::List => list_tasks(),
-    }
-}
-
-fn add_task(task: &str) {
-    let mut file = OpenOptions::new()
-        .create(true)
-        .append(true)
-        .open("todo.txt")
-        .expect("Failed to open file");
-
-    writeln!(file, "{}", task).expect("Failed to write to file");
-    println!("Added task: {}", task);
-}
-
-fn list_tasks() {
-    let content = read_to_string("todo.txt").expect("Failed to read file");
-
-    if content.is_empty() {
-        println!("No tasks found.");
-    } else {
-        for (index, task) in content.lines().enumerate() {
-            println!("{}: {}", index, task);
-        }
-    }
-}
\ No newline at end of file
+name: Rust
+
+on:
+  push:
+    branches: ["main"]
+  pull_request:
+    branches: ["main"]
+
+env:
+  CARGO_TERM_COLOR: always
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v4
+      - name: Build
+        run: cargo build --verbose --no-default-features
+      - name: Run tests
+        run: cargo test --verbose --no-default-features --lib
```

Comment:
Lets remove the `--no-default-features` from both and the `--lib` from the tests!

I doubt we will have people toggle feature flags, but if they do we should run tests for them!